lib/user.rb

```diff
@@ -0,0 +1,35 @@
+class User
+  attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(user, joke)
+    user.learn(joke)
+  end
+
+  def joke_by_id(id)
+    @jokes.select do |joke|
```

Comment:
`select` is an alias of `find_all`, which always returns an array. If we look at the interaction pattern, we don't want an array-- we want a single joke instance. So `detect` or its alias `find` would be better choices, since it gives you back the single instance. 

Nice use of an advanced enumerable though-- you had exactly the right idea, and it was almost the right tool!

Reply:
Thank-you! That's helpful!